### PR TITLE
Remove redundant calls to gtest_discover_tests with default args in AMQP tests

### PR DESCRIPTION
**Pull request by ahsonkhan**
Part of the investigation for https://github.com/Azure/azure-sdk-for-cpp/issues/5342

~Building on top of https://github.com/Azure/azure-sdk-for-cpp/pull/5631, to avoid seeing flakiness in CI, while investigating independent issues.~

### CODE REVIEW

File: sdk/core/azure-core-amqp/test/ut/link_tests.cpp [hunk 1]
```diff
@@ -286,7 +286,9 @@ namespace Azure { namespace Core { namespace Amqp { namespace Tests {
     listener.Stop();
   }
 
-  TEST_F(TestLinks, LinkAttachDetachMultipleOneSession)
+  // This test fails intermittently, with segfault:
+  // https://github.com/Azure/azure-sdk-for-cpp/issues/5536
+  TEST_F(TestLinks, DISABLED_LinkAttachDetachMultipleOneSession)
```
Comment (LarryOsterman) [inline sdk/core/azure-core-amqp/test/ut/link_tests.cpp:291]: This change does not appear to be related to the reported subject of the PR.

**Review by LarryOsterman — CHANGES_REQUESTED** [review]

File: sdk/core/azure-core-amqp/test/ut/link_tests.cpp (hunk 1 above)
Comment (ahsonkhan) [reply sdk/core/azure-core-amqp/test/ut/link_tests.cpp:291]: Trying to run CI a few times to see if this fixes it, and didn't want to get distracted by the other known issue:

> Building on top of https://github.com/Azure/azure-sdk-for-cpp/pull/5631, to avoid seeing flakiness in CI, while investigating independent issues.

**Review by LarryOsterman — APPROVED** [review]
